plugins/modules/ec2_vpc_vgw.py

```diff
@@ -270,50 +291,48 @@ def create_vgw(client, module):
         params["AmazonSideAsn"] = module.params.get("asn")
 
     try:
-        response = client.create_vpn_gateway(aws_retry=True, **params)
-        get_waiter(client, "vpn_gateway_exists").wait(VpnGatewayIds=[response["VpnGateway"]["VpnGatewayId"]])
+        response = create_vpn_gateway(client, **params)
+        get_waiter(client, "vpn_gateway_exists").wait(VpnGatewayIds=[response["VpnGatewayId"]])
     except botocore.exceptions.WaiterError as e:
-        module.fail_json_aws(
-            e, msg=f"Failed to wait for Vpn Gateway {response['VpnGateway']['VpnGatewayId']} to be available"
-        )
+        module.fail_json_aws(e, msg=f"Failed to wait for Vpn Gateway {response['VpnGatewayId']} to be available")
     except is_boto3_error_code("VpnGatewayLimitExceeded") as e:
         module.fail_json_aws(e, msg="Too many VPN gateways exist in this account.")
-    except (
-        botocore.exceptions.ClientError,
-        botocore.exceptions.BotoCoreError,
-    ) as e:  # pylint: disable=duplicate-except
-        module.fail_json_aws(e, msg="Failed to create gateway")
+    except AnsibleEC2Error as e:
+        module.fail_json_aws_error(e)
 
     result = response
     return result
 
 
-def delete_vgw(client, module, vpn_gateway_id):
+def delete_vgw(client, module: AnsibleAWSModule, vpn_gateway_id: str) -> Optional[str]:
     try:
-        response = client.delete_vpn_gateway(VpnGatewayId=vpn_gateway_id, aws_retry=True)
-    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
-        module.fail_json_aws(e, msg="Failed to delete gateway")
+        delete_vpn_gateway(client, vpn_gateway_id)
+    except AnsibleEC2Error as e:
+        module.fail_json_aws_error(e)
 
     # return the deleted VpnGatewayId as this is not included in the above response
     result = vpn_gateway_id
     return result
 
 
-def find_vpc(client, module):
+def find_vpc(client, module: AnsibleAWSModule) -> Optional[Any]:
     params = dict()
-    params["vpc_id"] = module.params.get("vpc_id")
+    vpc_id = module.params.get("vpc_id")
 
-    if params["vpc_id"]:
+    if vpc_id:
+        params["VpcIds"] = [vpc_id]
         try:
-            response = client.describe_vpcs(VpcIds=[params["vpc_id"]], aws_retry=True)
-        except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
-            module.fail_json_aws(e, msg="Failed to describe VPC")
+            response = describe_vpcs(client, **params)
+        except AnsibleEC2Error as e:
+            module.fail_json_aws_error(e)
 
     result = response
     return result
```

Comment:
Ad I would initialise response before.

```suggestion
    return response
```